packages/salesforce/src/Adaptor.js

```diff
@@ -783,6 +783,56 @@ export function toUTF8(input) {
   return anyAscii(input);
 }
 
+/**
+ * Send REST API request with given HTTP request info, with connected session information.
+ *
+ * @example
+ * request('/actions/custom/flow/POC_OpenFN_Test_Flow', {
+ *   method: 'POST',
+ *   body: { inputs: [{}] },
+ * });
+ * @param {String} path - URL to send HTTP request
+ * @param {Object} options - HTTP API request options
+ * @param {String} options.method - HTTP method URL to send HTTP request
+ * @param {Object} [options.headers] - HTTP request headers in hash object (key-value)
+ * @param {Object} [options.body] - HTTP request body
+ * @param {Function} callback - A callback to execute once the record is retrieved
+ * @returns {Operation}
+ */
+export function request(path, options, callback) {
+  return async state => {
+    const { connection } = state;
+    const [resolvedPath, resolvedOptions] = newExpandReferences(
+      state,
+      path,
+      options
+    );
+    const {
+      method = 'GET',
+      body,
+      headers = { 'content-type': 'application/json' },
+    } = resolvedOptions;
+
+    const requestOptions = {
+      url: resolvedPath,
+      method,
+      headers,
+      body: JSON.stringify(body),
+    };
+
+    await connection.request(requestOptions).then(response => {
+      const nextState = {
+        ...composeNextState(state, response),
```

Comment:
This doesn't look right.

The jsdocs aren't helpful, but it looks like the `request` function just returns the response body to us. In which case, there's no point in writing it to state root? So I think 
```
const nextState = composeNextState(state, response)
```

Reply:
And actually you should rename `response` to `result`

Reply:
If I'm wrong and the request really does return a response object, then we need to do some work to write `response.body` to `state.data`.